src/main/java/games/strategy/engine/data/TripleaHandler.java

```diff
@@ -0,0 +1,475 @@
+          handleDelegate(attributes);
+          break;
+        case XML_NODE_NAME_STEP:
+          handleStep(attributes);
+          break;
+        case "playerProduction":
+          handlePlayerProduction(attributes);
+          break;
+        case XML_NODE_NAME_COST:
+          handleCost(attributes);
+          break;
+        case XML_NODE_NAME_RESULT:
+          handleResult(attributes);
+          break;
+        case XML_NODE_NAME_FRONTIER_RULES:
+          handleFrontierRules(attributes);
+          break;
+        case XML_NODE_NAME_OPTION:
+          handleOption(attributes);
+          break;
+        case XML_NODE_NAME_TERRITORY_OWNER:
+          handleTerritoryOwner(attributes);
+          break;
+        case XML_NODE_NAME_UNIT_PLACEMENT:
+          handleUnitPlacement(attributes);
+          break;
+        case "resourceGiven":
+          handleResourceGiven(attributes);
+          break;
+        case XML_NODE_NAME_PROPERTY:
+          handleProperty(attributes);
+          break;
+        case XML_NODE_NAME_NUMBER:
+          handleNumber();
+          break;
+        case XML_NODE_NAME_BOOLEAN:
+          handleBoolean();
+          break;
+        case "tech":
+          handleTech(attributes);
+          break;
+        case XML_NODE_NAME_PRODUCTION_FRONTIER:
+          handleProductionFrontier(attributes);
+          break;
+        case XML_NODE_NAME_ATTACHMENT:
+          handleAttachment(attributes);
+          break;
+        case XML_NODE_NAME_VALUE:
+        case XML_NODE_NAME_SEQUENCE:
+        case XML_NODE_NAME_RESOURCE_LIST:
+        case XML_NODE_NAME_UNIT_LIST:
+        case "relationshipTypes":
+        case "territoryEffectList":
+        case "technology":
+        case XML_NODE_NAME_OWNER_INITIALIZE:
+        case XML_NODE_NAME_UNIT_INITIALIZE:
+        case "resouceInitialize":
+        case "technologies":
+        case "playerTech":
+        case "category":
+        default:
+          break;
+      }
+    } catch (GameParseException e) {
+      throw new SAXException(e);
+    }
+  }
+
+  private void checkParentNode(String qname) throws GameParseException {
+    if (!stack.isEmpty()) {
+      String actualName = stack.peek().getName();
+      Collection<String> parents = requiredParents.get(qname);
+      if (parents.isEmpty()) {
+        throw new GameParseException("Unknown element '" + qname + "'");
+      } else if (!parents.contains(actualName)) {
+        throw new GameParseException(
+            "Parent of '" + qname + "' must be one of " + Arrays.toString(parents.toArray()) + ", but was '"
+                + actualName + "'");
+      }
+    } else if (!qname.equals(XML_NODE_NAME_GAME)) {
+      throw new GameParseException("Root element must be 'game', found '" + qname + "' instead");
+    }
+  }
+
+  protected abstract void handleInfo(Attributes attributes) throws GameParseException;
+
+  protected abstract void handleLoader(Attributes attributes) throws GameParseException;
```

Comment:
I like the idea of having interfaces to do the individual jobs, this would be a pretty mayor change though, so not something I can do from here ^^
So this will have to wait until I'm back home unless @ssoloff wants to pick up my changes if he got too much time 😛